### PR TITLE
🌱 Remove `clusterctl backup` command and Backup function from Client interface

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

- Removes the `clusterctl backup` command
- Removes the `Backup` function in Client interface
- Removes tests
- Updates docs

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #7713

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Looks like the CI isn't running at all?

**Comment by oscr** [comment]
No, seems like there is an issue with CI.

**Comment by oscr** [comment]
apidiff failure is expected since it actually changes the api

```
   Incompatible changes:
  - ObjectMover.Backup: removed
sigs.k8s.io/cluster-api/cmd/clusterctl/client
  Incompatible changes:
  - (*clusterctlClient).Backup: removed
  - BackupOptions: removed
  - Client.Backup: removed 
```

**Review by Jont828 — COMMENTED** [review]
Thanks for the contribution! Everything seems reasonable to me, I don't see anything wrong that stands out. Are there any e2e tests that rely on this/would we need to update anything in test-infra?

**Comment by oscr** [comment]
> Thanks for the contribution! Everything seems reasonable to me, I don't see anything wrong that stands out. Are there any e2e tests that rely on this/would we need to update anything in test-infra?

Thanks @Jont828! I searched for it and the `pull-cluster-api-e2e-full-main` test passes so I hope it should be fine. apidiff test failure is expected since it changes :)

**Comment by ykakarap** [comment]
Overall looks good. I will take another pass to see if any of our test need to be adjusted for this.
And yes, the apidiff is expected.

**Review by ykakarap — APPROVED** [review]
/lgtm

**Review by fabriziopandini — COMMENTED** [review]
@oscr 
Thanks for working on this issue!
Let's keep the unit tests so we can exercise the clusterctl backup replacement (toDirectory)
Also could you please remove the comment about clusterctl backup from https://cluster-api.sigs.k8s.io/clusterctl/commands/commands.html
Oterwise lgtm for me

File: docs/book/src/clusterctl/commands/move.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ while doing the move operation, and possible race conditions happening while the
 remediating etc. has never been investigated nor addressed.
 
 In order to avoid further confusion about this point, `clusterctl backup` and `clusterctl restore` commands have been
```
Comment (oscr) [inline docs/book/src/clusterctl/commands/move.md:54]: I thought about if I should remove this whole section or just change deprecated to deleted. However I kept it as a reference. But I think it should definitely be deleted in the future.

Comment (fabriziopandini) [reply docs/book/src/clusterctl/commands/move.md:54]: Nice catch, ok to leave this for now

File: cmd/clusterctl/client/move_test.go [hunk 2]
```diff
@@ -139,69 +139,6 @@ func Test_clusterctlClient_Move(t *testing.T) {
 	}
 }
 
-func Test_clusterctlClient_Backup(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply cmd/clusterctl/client/move_test.go:142]: It seems this still need to be changed

[the remark this answers is not in the document]
Comment (oscr) [reply cmd/clusterctl/client/move_test.go:142]: Sorry I missed this one.

**Comment by fabriziopandini** [comment]
/lgtm
/approve

Thanks for this cleanup! next up is restore!

**Pull request merged**